presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayJoin.java

```diff
@@ -315,15 +315,24 @@ public static void arrayJoinProvidedBlock(
             Slice nullReplacement)
     {
         int numElements = arrayBlock.getPositionCount();
+        boolean firstElement = true;
```

Comment:
```suggestion
        boolean hasOutput = false;
```

I think this name is a bit misleading.  It seems this is actually referring to whether or not this function has encountered any non-null positions in the block, or if there has been a null position, that it hasn't been replaced with the replacement string, and in either case has this value been written to the block builder.  Hence I'd rename this to something like the above.